### PR TITLE
Fix check is crawlera response

### PR DESCRIPTION
**Pull request by denisgermano**

### CODE REVIEW

**Comment by eLRuLL** [comment]
very nice work, just a question here, is Crawlera deprecating the `X-Crawlera-Version` header? If so, could you please point that out here and add it to the documentation?

This should definitely be added as a new version as well, so we are missing the bump.

**Comment by denisgermano** [comment]
Hey @eLRuLL It is not being deprecated, the latest release has a bug that is returning an empty value for the `X-Crawlera-Version` header

File: .travis.yml [hunk 1]
```diff
@@ -5,12 +5,14 @@ matrix:
...
-      env: TOXENV=py36,stack-scrapy-1.3-py3,stack-scrapy-1.4-py3,stack-scrapy-1.5-py3
+    - python: 3.5.2
+      env: TOXENV=py35,stack-scrapy-1.8-py3,stack-scrapy-2.0-py3,stack-scrapy-2.1-py3,stack-scrapy-2.2-py3,stack-scrapy-2.3-py3
+    - python: 3.6.1
+      env: TOXENV=py36,stack-scrapy-2.2-py3,stack-scrapy-2.3-py3,stack-scrapy-2.4-py3
+    - python: 3.7
+      env: TOXENV=py37,stack-scrapy-2.2-py3,stack-scrapy-2.3-py3,stack-scrapy-2.4-py3
     - python: 3.8
-      env: TOXENV=docs
+      env: TOXENV=py38,docs,stack-scrapy-2.2-py3,stack-scrapy-2.3-py3,stack-scrapy-2.4-py3
```
Comment (dangra) [inline .travis.yml:15]: I think we need a different PR for this test matrix updates.

File: docs/news.rst [hunk 2]
```diff
@@ -3,6 +3,11 @@
 Changes
 =======
 
+v1.7.1 (2020-10-22)
+-------------------
+- Consider Crawlera response if contains any `X-Crawlera-` header
+- Build the documentation in Travis CI and fail on documentation issues
```
Comment (dangra) [inline docs/news.rst:9]: Nice but release notes shouldn't be part of this PR either.

File: scrapy_crawlera/middleware.py [hunk 3]
```diff
@@ -283,7 +283,7 @@ def _get_url_domain(self, url):
         return parsed.netloc
 
     def _is_crawlera_response(self, response):
-        return bool(response.headers.get("X-Crawlera-Version"))
+        return self._has_crawlera_headers(response)
```
Comment (dangra) [inline scrapy_crawlera/middleware.py:286]: `_has_crawlera_headers` is O(n) while current is O(1). I think we should stick to:


```suggestion
        return bool("X-Crawlera-Version" in response.headers)
```

**Review by dangra — CHANGES_REQUESTED** [review]
The PR must be split before the functionality changes are merged.

File: .travis.yml (hunk 1 above)
Comment (denisgermano) [reply .travis.yml]: https://github.com/scrapy-plugins/scrapy-crawlera/pull/90

File: scrapy_crawlera/middleware.py (hunk 3 above)
Comment (denisgermano) [reply scrapy_crawlera/middleware.py]: Done
